[CI] VectorIT testFilteredQueryStrategy failing

**Build Scans:**
- [elasticsearch-periodic #11805 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/xknbkbdpv4nke)
- [elasticsearch-intake #33771 / part1](https://gradle-enterprise.elastic.co/s/mhnmhumt22ptm)
- [elasticsearch-pull-request #116451 / part-1](https://gradle-enterprise.elastic.co/s/2quc6ropadv2y)
- [elasticsearch-periodic-platform-support #11752 / debian-13_platform-support-unix](https://gradle-enterprise.elastic.co/s/zug3er3x3xks6)
- [elasticsearch-periodic-platform-support #11752 / sles-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/4i25plhirnbhw)

**Reproduction Line:**
```
./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.search.query.VectorIT.testFilteredQueryStrategy" -Dtests.seed=225C2B3D9E8A20BB -Dtests.jvm.argline="-Des.concurrent_search=true" -Dtests.locale=ar-DJ -Dtests.timezone=Africa/Tunis -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.search.query.VectorIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testFilteredQueryStrategy'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: fanoutVectorOps [31] is not gt acornVectorOps [33]
```

**Issue Reasons:**
- [main] 5 failures in test testFilteredQueryStrategy (0.7% fail rate in 680 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -312,9 +312,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
   method: test {csv-spec:spatial_shapes.CartesianPolygonEnvelopeXYMinMax}
   issue: https://github.com/elastic/elasticsearch/issues/141425
-- class: org.elasticsearch.search.query.VectorIT
-  method: testFilteredQueryStrategy
-  issue: https://github.com/elastic/elasticsearch/issues/141469
 - class: org.elasticsearch.xpack.security.authc.TokenServiceTests
   method: testAttachAndGetToken
   issue: https://github.com/elastic/elasticsearch/issues/141470

--- a/server/src/internalClusterTest/java/org/elasticsearch/search/query/VectorIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/search/query/VectorIT.java
@@ -74,6 +74,13 @@ public class VectorIT extends ESIntegTestCase {
     }
 
     public void testFilteredQueryStrategy() {
+        // Disable early termination to isolate the filter heuristic behavior
+        client().admin()
+            .indices()
+            .prepareUpdateSettings(INDEX_NAME)
+            .setSettings(Settings.builder().put(DenseVectorFieldMapper.HNSW_EARLY_TERMINATION.getKey(), false))
+            .get();
+
         float[] vector = new float[16];
         randomVector(vector, 25);
         int upperLimit = 35;

